File: HW/HW8/longPal.java
package HW8;

import java.util.HashMap;

public class longPal {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> freq = new HashMap<>();
        for (char c : s.toCharArray()) {
            freq.put(c, freq.getOrDefault(c, 0) + 1);
        }
        int len = 0;
        boolean oddF = false;
        

        for (int x : freq.values()) {
            if (x % 2 == 0) {


                len += x;
            } else {
                len += x - 1;

                oddF = true;
            }
        }
        if (oddF) {
            len++;
        }
        return len;
    }

    public static void main(String[] args) {
        longPal lp = new longPal();
        String s = "abccccdd"; 
        String s1 = "speediskey";

        System.out.println(lp.longestPalindrome(s1));
        System.out.println(lp.longestPalindrome(s)); 
    }
}

File: HW/HW8/longPal.py
class longPal:
    def longestPalindrome(self, s: str) -> int:
        freq = {}
        
        for char in s:
            freq[char] = freq.get(char, 0) + 1
        
        len = 0
        oddF = False
        
        for freq in freq.values():
            if freq % 2 == 0:
                len += freq
            else:
                len += freq - 1
                oddF = True
        
        if oddF:
            len += 1
        
        return len


longPal = longPal()
s1 = "abccccdd"
s2 = "speediskey"
print(s1)
print(longPal.longestPalindrome(s1))
print(s2)
print(longPal.longestPalindrome(s2))
